rescompy/rescompy.py

```diff
@@ -209,65 +208,126 @@ class TrainResult:
         feature_function (Callable): The function that transforms the states
             and inputs to features.
         weights (np.ndarray): The array of output weights.
-        transient_length (int): The length of the initial transient.
+        transient_lengths (List[int]): The lengths of the initial transients.
     """
 
     @validate_arguments(config=dict(arbitrary_types_allowed=True))
     def __init__(
         self,
-        states:           np.ndarray,
-        inputs:           np.ndarray,
-        target_outputs:   np.ndarray,
-        feature_function: Callable,
-        weights:          np.ndarray,
-        transient_length: int,
-        jacobian:         Optional[np.ndarray] = None
+        states:            List[np.ndarray],
+        inputs:            List[np.ndarray],
+        target_outputs:    Union[List[None],  List[np.ndarray]],
+        transient_lengths: List[int],
+        accessible_drives: List[int],
+        feature_function:  Callable,
```

Comment:
Should it be

`feature_function:  Union[features.ESNFeature, Callable],`

Reply:
Yes, I'll change that.